code/game/objects/structures/crates_lockers/crates.dm

```diff
@@ -133,6 +139,21 @@
 		if(elevation)
 			AddElement(/datum/element/elevation, pixel_shift = elevation)
 
+///Applies a proc when movable enters the turf of our crate
+/obj/structure/closet/crate/proc/on_entered(datum/source, atom/movable/soapbox_arrive)
+	SIGNAL_HANDLER
+	if(!opened)
+		RegisterSignal(soapbox_arrive, COMSIG_MOB_SAY, PROC_REF(soapbox_speech))
```

Comment:
If I'm not mistaken you'll need to add code for the box moving out from underneath the mob, which should just be registering a `moved` signal for the mob to, well, unregister the say signal (and the moved signal itself)

Reply:
Should be fixed now